### ISSUE TITLE
Sorting on a multi-valued struct field value incorrectly triggers sorting optimization via match-phase degradation 

### ISSUE THREAD

**Issue by jobergum** [vespa-engine/vespa#31564]
TL;DR

Match phase degradation is triggered when sorting on a array of struct field - triggering the following response:

```The attribute 'array_field.struct_field_name' is not available for match-phase. It must be a single value numeric attribute with fast-search.```

## How to reproduce 

Extended `music.sd` schema with a struct foo type and an array of long for comparison:

```
schema music {

    struct foo {
        field timestamp type long{}
	}

    document music {

        field artist type string {
            indexing: summary | index
        }

        field album type string {
            indexing: summary | index
            index: enable-bm25
        }

        field year type int {
            indexing: summary | attribute
        }

        field category_scores type tensor<float>(cat{}) {
            indexing: summary | attribute
        }

        field timestamps type array<long> {
            indexing: summary|attribute
            attribute: fast-search
        }

        field foos type array<foo> {
            indexing: summary
            struct-field timestamp {
                indexing: attribute
                attribute: fast-search
	    }
	}
    }
```

```json
{
    "put": "id:mynamespace:music::a-head-full-of-dreams",
    "fields": {
        "album": "A Head Full of Dreams",
        "artist": "Coldplay",
        "year": 2015,
        "category_scores": {
            "pop": 1,
            "rock": 0.2,
            "jazz": 0
        },
        "foos": [
            {
                "timestamp": 12345678910
            }
        ],
        "timestamps": [
            12345678910
        ]
    }
}
```

### Query 
Works as expected, no sort optimization via match phase is triggered:
```
vespa query 'yql=select * from music where true order by timestamps'
```

Fails
```
vespa query 'yql=select * from music where true order by foos.timestamp'
```

Response:
```
{
    "root": {
        "id": "toplevel",
        "relevance": 1.0,
        "fields": {
            "totalCount": 0
        },
        "errors": [
            {
                "code": 4,
                "summary": "Invalid query parameter",
                "source": "music",
                "message": "The attribute 'foos.timestamp' is not available for match-phase. It must be a single value numeric attribute with fast-search."
            }
        ]
    }
}
```

Disabling sorting degradation and the query works as expected
```
vespa query 'yql=select * from music where true order by foos.timestamp' 'sorting.degrading=false'
```

**Comment by baldersheim** [comment]
Suggested remediation #31565

**Comment by baldersheim** [comment]
Fixed in 8.358.15
Problem was that if the struct was used as array elements or map elements the information that it was multivalue was not properly propagated.
